src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java

```diff
@@ -89,32 +89,33 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
                             refIds.size());
                     }
                 }
+            }
 
-                if (numFilesSeen != uniqueRefIds.size()) {
-                    logger.info("some files are repeated. {} unique values across {} total file references",
-                        uniqueRefIds.size(), numFilesSeen);
-                }
-                List<String> uniqueRefIdsAsList = new ArrayList<>(uniqueRefIds);
-                Dataset dataset = datasetService.retrieve(snapshotSource.getDataset().getId());
-
-                String addFilesTimer = performanceLogger.timerStart();
-                fileDao.addFilesToSnapshot(dataset, snapshot, uniqueRefIdsAsList);
-                performanceLogger.timerEndAndLog(
-                    addFilesTimer,
-                    context.getFlightId(),
-                    this.getClass().getName(),
-                    "fileDao.addFilesToSnapshot",
-                    uniqueRefIds.size());
-
-                String addDependenciesTimer = performanceLogger.timerStart();
-                dependencyDao.storeSnapshotFileDependencies(dataset, snapshot.getId().toString(), uniqueRefIdsAsList);
-                performanceLogger.timerEndAndLog(
-                    addDependenciesTimer,
-                    context.getFlightId(),
-                    this.getClass().getName(),
-                    "dependencyDao.storeSnapshotFileDependencies",
-                    uniqueRefIds.size());
+            if (numFilesSeen != uniqueRefIds.size()) {
+                logger.info("some files are repeated. {} unique values across {} total file references",
+                    uniqueRefIds.size(), numFilesSeen);
             }
+
+            List<String> uniqueRefIdsAsList = new ArrayList<>(uniqueRefIds);
```

Comment:
We should change our code to not use apache partition, which requires `List<T>` instead of `Collection<T>`. This can be done in JDK 8 in a few lines and will support `Collection`.

Reply:
Ooh, yeah, I might look into switching that at some point